lib/node_modules/@stdlib/repl/lib/completer_engine.js

```diff
@@ -0,0 +1,693 @@
+
+			// Move the cursor to the start of completion prefix:
+			self._rli.cursor -= self._completionPrefix.length;
+
+			// Write the auto-completion string:
+			self._rli.write( autoCompletion );
+
+			// Resume the input stream:
+			self._rli.resume();
+			return;
+		}
+		debug( 'No auto-completion candidate, displaying all possible completions.' );
+
+		// Check if completions can be displayed...
+		if ( !self._isDisplayable() ) {
+			debug( 'TTY height too short. exiting completer...' );
+			self._rli.resume();
+			return;
+		}
+
+		// Display completions:
+		self._displayCompletions();
+		self._isNavigating = true;
+
+		// Resume the input stream:
+		self._rli.resume();
+	}
+});
+
+/**
+* Displays the completions to the output stream.
+*
+* @private
+* @name _displayCompletions
+* @memberof CompleterEngine.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, '_displayCompletions', function displayCompletions() {
+	var columns;
+	var rows;
+	var dy;
+	var i;
+
+	// Determine number of columns of completions that should be displayed to the output stream
+	this._completionsLength = [];
+	for ( i = 0; i < this._completionsList.length; i++ ) {
+		this._completionsLength.push( this._completionsList[ i ].length );
+	}
+	this._widthOfColumn = maxInArray( this._completionsLength.length, this._completionsLength, 1 ) + 4; // eslint-disable-line max-len
+
+	// Highlight completions if operating in "terminal" mode...
+	if ( this._repl._isTTY ) {
+		this._highlightedCompletions = this._highlightCompletions();
+	} else {
+		this._highlightedCompletions = this._completionsList;
+	}
+
+	// Determine dimensions of the output grid:
+	columns = this._completionColumns();
+	rows = this._completionRows( columns );
+
+	// Move cursor to the output row:
+	dy = this._multiline.inputHeight() - this._multiline.lineIndex() - 1;
+	readline.moveCursor( this._ostream, 0, dy );
+
+	// Write completions to the output stream:
+	this._ostream.write( this._drawOutput( rows, columns ) );
+
+	// Bring the cursor back to the current line:
+	readline.moveCursor( this._ostream, 0, -1 * ( rows + this._multiline.inputHeight() - this._multiline.lineIndex() ) ); // eslint-disable-line max-len
+	readline.cursorTo( this._ostream, this._rli.cursor + this._repl.promptLength() ); // eslint-disable-line max-len
+});
+
+/**
+* Draws the completions output grid.
+*
+* @private
+* @name _drawOutput
+* @memberof CompleterEngine.prototype
+* @type {Function}
+* @param {number} rows - number of rows in output grid
+* @param {number} columns - number of columns in output grid
+* @returns {string} output string
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, '_drawOutput', function drawOutput( rows, columns ) {
+	var whitespaces;
+	var completion;
+	var lineIndex;
+	var output;
+	var i;
+
+	// Draw the output grid:
+	output = '\r\n';
```

Comment:
Why are you using Windows CRLF endings, rather than Unix? Typically, we'd just write `\n`.

Reply:
When I wrote this, I was following Node readline's built in completer as a reference. IIRC they use this. I'll change it to just `\n`

Reply:
It's possible we should be more platform-aware, but, IIRC, we use `\n` everywhere else in the REPL, so should be consistent.